src/Nethermind/Nethermind.State/PreBlockCaches.cs

```diff
@@ -5,28 +5,101 @@
 using System.Collections.Concurrent;
 using System.Numerics;
 using System.Runtime.InteropServices;
-
+using System.Threading.Tasks;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
+using Nethermind.Core.Collections;
+using Nethermind.Logging;
 using Nethermind.Trie;
 
 namespace Nethermind.State;
 
 public class PreBlockCaches
 {
-    public ConcurrentDictionary<StorageCell, byte[]> StorageCache { get; } = new(Environment.ProcessorCount * 2, 4096 * 4);
-    public ConcurrentDictionary<AddressAsKey, Account> StateCache { get; } = new(Environment.ProcessorCount * 2, 4096 * 4);
-    public ConcurrentDictionary<NodeKey, byte[]?> RlpCache { get; } = new(Environment.ProcessorCount * 2, 4096 * 4);
-    public ConcurrentDictionary<PrecompileCacheKey, (ReadOnlyMemory<byte>, bool)> PrecompileCache { get; } = new(Environment.ProcessorCount * 2, 4096 * 4);
+    private const int InitialCapacity = 4096 * 8;
+    private static int LockPartitions => CollectionExtensions.LockPartitions;
+
+    private readonly ILogger _logger;
+    private readonly Action _clearStorageCache;
+    private readonly Action _clearStateCache;
+    private readonly Action _clearRlpCache;
+    private readonly Action _clearPrecompileCache;
+
+    private ConcurrentDictionary<StorageCell, byte[]> _storageCache = new(LockPartitions, InitialCapacity);
+    private ConcurrentDictionary<AddressAsKey, Account> _stateCache = new(LockPartitions, InitialCapacity);
+    private ConcurrentDictionary<NodeKey, byte[]?> _rlpCache = new(LockPartitions, InitialCapacity);
+    private ConcurrentDictionary<PrecompileCacheKey, (ReadOnlyMemory<byte>, bool)> _precompileCache = new(LockPartitions, InitialCapacity);
+
+    public PreBlockCaches(ILogManager logManager)
+    {
+        _logger = logManager?.GetClassLogger<WorldState>() ?? throw new ArgumentNullException(nameof(logManager));
+        _clearStorageCache = () => _storageCache.NoResizeClear();
+        _clearStateCache = () => _stateCache.NoResizeClear();
+        _clearRlpCache = () => _rlpCache.NoResizeClear();
+        _clearPrecompileCache = () => _precompileCache.NoResizeClear();
+    }
+
+    public ConcurrentDictionary<StorageCell, byte[]> StorageCache => _storageCache;
+    public ConcurrentDictionary<AddressAsKey, Account> StateCache => _stateCache;
+    public ConcurrentDictionary<NodeKey, byte[]?> RlpCache { get; } = new(LockPartitions, InitialCapacity);
+    public ConcurrentDictionary<PrecompileCacheKey, (ReadOnlyMemory<byte>, bool)> PrecompileCache { get; } = new(LockPartitions, InitialCapacity);
```

Comment:
We have a property and a field initialized with `new`? Probably a mistake.

Reply:
"The fix was a lie"

Yes mistake 😅